multimodal/src/autogluon/multimodal/models/ft_transformer.py

```diff
@@ -594,12 +595,17 @@ def __init__(
             initialization="uniform",
         )
 
-        # init weights
+        # init tokenizer and head weights
         if self.numerical_feature_tokenizer:
             self.numerical_adapter.apply(init_weights)
         if self.categorical_feature_tokenizer:
             self.categorical_adapter.apply(init_weights)
         self.head.apply(init_weights)
+        # init transformer backbone from provided checkpoint
+        if checkpoint_name:
+            ckpt = torch.load(checkpoint_name)
+            self.transformer.load_state_dict(ckpt["state_dict"])
```

Comment:
This only loads weights for `self.transformer`. What if the saved weights also have `self.categorical_adapter`, `self.numerical_adapter`, and `self.head`?

Reply:
Currently, this is mostly for the [XTab pretraining](https://github.com/BingzhaoZhu/XTab). And the use case is that users load XTab pre-trained ft-transformer weights and do finetuning.

Reply:
Can we add tests?

Reply:
Added unit test. The [XTab](https://github.com/BingzhaoZhu/XTab) repo has the benchmarking results.

Reply:
The readme in the XTab repo only shows the result on one toy dataset. Benchmark results on all tabular datasets are unclear. It's also unclear whether we can reproduce the results in the paper due to some details like light finetuning vs heavy finetuning.